.github/workflows/read-size.yml

```diff
@@ -0,0 +1,50 @@
+name: Read size
+
+on:
+  pull_request:
+    paths:
+      - 'src/**'
+      - 'package.json'
+
+# This workflow runs in a read-only environment. We can safely checkout
+# the PR code here.
+# Reference:
+# https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+permissions:
+  contents: read
+
+jobs:
+  read-size:
+    name: Tree-shaking
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v3
+      - name: Install Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'npm'
+      - name: Install dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build-module
+      - name: === Test tree-shaking ===
+        run: npm run test-treeshake
+      - name: Read bundle sizes
+        id: read-size
+        run: |
+          FILESIZE=$(stat --format=%s test/treeshake/three.module.min.js)
+          gzip -k test/treeshake/three.module.min.js
```

Comment:
Module isn't shipped minified.  See next comment.

Reply:
The size reported is the size of the bundle minified. I didn't use `three.min.js` because that file is deprecated.

Reply:
Tools usually report the minified size and that's what I did here. I think this is what @mrdoob meant when he asked for the bundle size to be reported.

Reply:
Tools report the minimised size when minimizing has some meaning.  Applying minification to a file that can never ship minimised in this way is not helpful.

Reply:
No. It can be optimised slightly in the packaging process but not as aggressively without losing the benefits of using ESM.

Reply:
Tools usually report a minified size of the minified size.  They don't create a random build type, minimise it, and report on it.  When tools like vite report on the ESM bundle size for a library, they are not reporting on the minimised size.  They only report minimised sizes when the target is an application.

Reply:
> No. It can be optimised slightly in the packaging process but not as aggressively without losing the benefits of using ESM.

The only thing that couldn't be minimised in ESM file are imports and exports which are negligible for a large library.

Reply:
That is true of the final bundler doing the minification sure, anything is possible.  What you are pointing out makes tree-shaking very difficult and sub optimal.  It works but with bad outcomes.

Seriously, I'm getting the feeling you will take any position opposite to me with little thought.  Please accept my apology if I have created this situation.  My goal is only to help and collaborate.

Reply:
Reporting a minified, compressed size is accepted practice in JavaScript packages, because that's the number that matters to any consumer who cares about their bundle size. Let's close out that portion of the discussion and resolve this thread. I'll address the "random build type" bit in another comment.